### ISSUE TITLE
Kernel keeps crashing

### ISSUE THREAD

**Issue by 211sk** [janboone/msc_datascience#24]
EDIT: By pruning and setting max depth, I was able to run the command. Thank you
Hello, I am trying to run a decision tree classifier but the kernel keeps crashing at this code. I was using google colab for this, and felt this could be a RAM issue so I changed the runtime settings to GPU and it continued. I shifted to Jupyterlab, and I am still getting the same result. 
Could you have a look at the code? Since I do not have any error, I don't know where exactly to look. The dataset is a small size 21MB dataset. Shape - (43662, 17)

'# training the decision tree classifier

from sklearn.tree import DecisionTreeClassifier

tree = DecisionTreeClassifier(random_state=1).fit(features_train, target_train)
print("Accuracy on training set: {:.3f}".format(tree.score(features_train, target_train)))
print("Accuracy on test set: {:.3f}".format(tree.score(features_test, target_test)))'

If there's something else I need to look into, please let me know. Thanking you in advance.
